src/praisonai-agents/praisonaiagents/agent/tool_execution.py

```
@@ -954,3 +958,97 @@ async def check_pending_approvals(self) -> Dict[str, Any]:
+        
+        Args:
+            function_name: Bridge tool name
+            arguments: Arguments passed to bridge tool
+            tool_call_id: Optional tool call ID
+            
+        Returns:
+            Result of bridge tool execution or unwrapped real tool call
+        """
+        # Ensure tool search metadata is available
+        if not hasattr(self, '_tool_search_metadata') or self._tool_search_metadata is None:
+            return f"Tool search not available or not in bridge mode"
```

Ruff (0.15.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)